osu.Game/Screens/Menu/DailyChallengeButton.cs

```diff
@@ -148,6 +152,9 @@ private void dailyChallengeChanged(ValueChangedEvent<DailyChallengeInfo?> _)
 
                 roomRequest.Success += room =>
                 {
+                    // force showing intro on the first time when a new daily challenge is up.
+                    statics.SetValue(Static.DailyChallengeIntroPlayed, false);
```

Comment:
I'm not sure this reset works as intended.

1. Log in
2. Go to daily challenge, intro plays
3. Exit daily challenge, log out and back in again
4. Go to daily challenge, intro plays again

This is happening because daily challenge room ID resets to null when logging out and is received from the server afresh on login.

To make this work you probably want to store the room ID here or something.

Reply:
I think it might be fine for it to play again after a logout / login.

Imagine the case of a shared PC where osu! is left running. I'd expect each user to see it once.

Reply:
Sure, but as mentioned this will also happen on connection drop-outs which is not desired.